per email from Memphis re step 6.1 generate Xcode project

pteeson opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

pteeson committed:
per email from Memphis re step 6.1 generate Xcode project - following commits will be squashed

pteeson committed:
per email from Memphis re step 6.1 generate Xcode project

pteeson committed:
# This is a combination of 3 commits.
# The first commit's message is:
per email from Memphis re step 6.1 generate Xcode project - following commits will be squashed

# The 2nd commit message will be skipped:

#	Step 6.1 Build with Xcode needs 3 corrections:
#	2 are in the instructions for building the Xcode 64bit and 32bit project.
#	Presently the location for the cmake result is ../kodi (i.e. the parent directory of kodi)
#	This is wrong per Memphis' email. The correct location should be .. (i.e. $HOME)
#	This also requires a change to the line beginning **Start Xcode... changing '$HOME/kodi-build' to '$HOME'

# The 3rd commit message will be skipped:

#	As per Rechi comment: In the 64bit and 32bit Xcode Generate project lines
#	remove incorrect ... and restore ```

Rechi commented:
@pteeson please give the commit a suitable commit message, which describe the changes. The current one doesn't give any clue what was changed and one always have to look at the actual changes to know.

Memphiz commented:
For changing the commit message do the rebase with “HEAD~1” this time and replace the word “pick” with “r” (reword) ... then :wq and change to commit message in the next editor session that follows ... after that push -f to your branch to update the PR.

pteeson committed:
Restore document to the way it was

pteeson committed:
Step 6.1 Build with Xcode needs 3 corrections:
2 are in the instructions for building the Xcode 64bit and 32bit project.
Presently the location for the cmake result is ../kodi (i.e. the parent directory of kodi)
This is wrong per Memphis' email. The correct location should be .. (i.e. $HOME)
This also requires a change to the line beginning **Start Xcode... changing '$HOME/kodi-build' to '$HOME'

pteeson commented:
After much fasting, sackcloth and ashes I think things may be correct. If not let me know.
> On Oct 7, 2018, at 4:57 AM, Memphiz <notifications@github.com> wrote:
> For changing the commit message do the rebase with “HEAD~1” this time and replace the word “pick” with “r” (reword) ... then :wq and change to commit message in the next editor session that follows ... after that push -f to your branch to update the PR.

Rechi commented on docs/README.macOS.md:
````diff
@@ -159,13 +159,13 @@ cd $HOME/kodi-build
 Generate Xcode project for 64bit (**recommended**):
 ```
-/Users/Shared/xbmc-depends/x86_64-darwin17.5.0-native/bin/cmake -G Xcode -DCMAKE_TOOLCHAIN_FILE=/Users/Shared/xbmc-depends/macosx10.13_x86_64-target-debug/share/Toolchain.cmake ../kodi
-```
````
don't change ***```*** to `...`

Rechi commented on docs/README.macOS.md:
````diff
@@ -159,13 +159,13 @@ cd $HOME/kodi-build
 Or generate Xcode project for 32bit:
 ```
-/Users/Shared/xbmc-depends/x86_64-darwin17.5.0-native/bin/cmake -G Xcode -DCMAKE_TOOLCHAIN_FILE=/Users/Shared/xbmc-depends/macosx10.13_i386-target-debug/share/Toolchain.cmake ../kodi
-```
````
same

pteeson committed:
As per Rechi comment: In the 64bit and 32bit Xcode Generate project lines
remove incorrect ... and restore ```

Memphiz commented:
Not quiet ther yet. Your PR has 4 commits now.
I suggest - change the things I marked and commit those.
You have 5 commits then.

Then squash those five commits down into a single one and reword the left commit
This can be done in one step then:
git rebase HEAD~5 -i
Editor opens:
First line - replace “pick” with “r” (for reword)
Replace “pick” of the other 4 commits with “f” (for fixup/squash)

:wq 
Editor opens again:
Reword the commit message
:wq

Then force push your branch.

If all is done right this PR will reflect the changes.

pteeson committed:
Step 6.1 Build with Xcode has 64bit & 32bit Generate Xcode project commands
Each of them has ../kodi as the destination. But this is wrong.
They have both been changed to .. as per Memphis' email

pteeson commented:
FWIW Nano is my global editor and it opens OK in the various commands.
It seems I have the non-Midas touch. I’m so sorry for the trouble I am causing. 
I really want to totally understand git in a team environment. Still a long way to go.

Here is the log as I tried to follow your steps:

Last login: Tue Oct  9 21:28:28 on ttys000
Gandalf:~ pteeson$ cd /Volumes/Data-b/kodi
Gandalf:kodi pteeson$ git status
On branch macOS-README-6.1
Your branch is up-to-date with 'origin/macOS-README-6.1'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   docs/README.macOS.md

no changes added to commit (use "git add" and/or "git commit -a")
Gandalf:kodi pteeson$ git add .
Gandalf:kodi pteeson$ git commit
[macOS-README-6.1 5a403a3] Step 6.1 Build with Xcode has 64bit & 32bit Generate Xcode project commands Each of them has ../kodi as the destination. But this is wrong. They have both been changed to .. as per Memphis' email
 1 file changed, 1 insertion(+), 1 deletion(-)
Gandalf:kodi pteeson$ git rebase HEAD~5 -i
[detached HEAD 6234051] per email from Memphis re step 6.1 generate Xcode project The 64bit and 32bit generate commands end with a destination of ../kodi. This is wrong They should be .. This fixes them.
 Date: Sat Oct 6 17:00:10 2018 -0400
 1 file changed, 2 insertions(+), 2 deletions(-)
You asked to amend the most recent commit, but doing so would make
it empty. You can repeat your command with --allow-empty, or you can
remove the commit entirely with "git reset HEAD^".
rebase in progress; onto 6d1afec
You are currently rebasing branch 'macOS-README-6.1' on '6d1afec'.

No changes

Could not apply 9ad1e6ac1c93d88a94419b4af9fdef4a9ad1aac3... Restore document to the way it was

Gandalf:kodi pteeson$ 
> On Oct 10, 2018, at 1:11 AM, Memphiz <notifications@github.com> wrote:
> 
> Not quiet ther yet. Your PR has 4 commits now.
> I suggest - change the things I marked and commit those.
> You have 5 commits then.
> 
> Then squash those five commits down into a single one and reword the left commit
> This can be done in one step then:
> git rebase HEAD~5 -i
> Editor opens:
> First line - replace “pick” with “r” (for reword)
> Replace “pick” of the other 4 commits with “f” (for fixup/squash)
> 
> :wq
> Editor opens again:
> Reword the commit message
> :wq
> 
> Then force push your branch.
> 
> If all is done right this PR will reflect the changes.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/xbmc/xbmc/pull/14532#issuecomment-428440020>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AF_MQjoWdge3QmTzWMfpEESA2ckeWdwZks5ujYGFgaJpZM4XLj8i>.
>

pteeson commented:
Would it not be simpler to delete/remove all of the 5 commits and start afresh?

Razzeee commented:
If all else fails we can even squash the commits for you

On Wed, 10 Oct 2018, 18:49 pteeson, <notifications@github.com> wrote:

> Would it not be simpler to delete/remove all of the 5 commits and start
> afresh?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/xbmc/xbmc/pull/14532#issuecomment-428790173>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AFqyZMezZW1U2qaEuqoaRr0RV3M_0mQ3ks5ujqOegaJpZM4XLj8i>
> .
>

Memphiz commented:
You have locally uncommitted changes that make my assumption mood.

You do the following now to start fresh:

git stash
git fetch upstream
git reset —hard upstream/master
Now make the change to the readme file
git commit -a -m “[docs] Fixed 2 typos in macOS readme”
git push -f origin macOS-README-6.1

pteeson commented:
OK good people that didn’t work- (I didn’t do the push because…). 
So here’s what I plan to do:
(0) Really understand the plumbing (Ch 10). I’m tired of not knowing what I’m doing.
 In parallel with (0):
(1) Ask you to delete from upstream any pushed items from me that have not been merged
(2) I will delete, when you say it’s ok, my forked Kodi and make a new one 
(3) Clone the fork to my local machine setting up the config, and a topic a.k.a. branch.
(4) Do the edit to the 6.1 Generate Xcode project as per Memphis’ email
(5) Buy everyone of you a virtual beverage of your choice to thank you for your patience & kindness.

> On Oct 11, 2018, at 1:05 AM, Memphiz <notifications@github.com> wrote:
> You have locally uncommitted changes that make my assumption mood.
> 
> You do the following now to start fresh:
> 
> git stash
> git fetch upstream
> git reset —hard upstream/master
> Now make the change to the readme file
> git commit -a -m “[docs] Fixed 2 typos in macOS readme”
> git push -f origin macOS-README-6.1
>

pteeson commented:
Sorry for the delay but I did extensive reading and experimenting.
I think my understanding is much better now.

OK as far as I can tell things are cleaned up here at origin and local. 
So I did this:

Gandalf:kodi pteeson$ git status
On branch macOS-README-6.1
Your branch is ahead of 'origin/macOS-README-6.1' by 1 commit.
  (use "git push" to publish your local commits)
nothing to commit, working directory clean
Gandalf:kodi pteeson$ git push
Counting objects: 4, done.
Delta compression using up to 16 threads.
Compressing objects: 100% (4/4), done.
Writing objects: 100% (4/4), 492 bytes | 0 bytes/s, done.
Total 4 (delta 3), reused 0 (delta 0)
remote: Resolving deltas: 100% (3/3), completed with 3 local objects.
To git@github.com:pteeson/xbmc.git
   480e904..b4648a1  macOS-README-6.1 -> macOS-README-6.1

Please let me know if it seems OK.

thanks...

Memphiz commented:
Well the resulting readme is fine but that history with 5 commits is not. We still would like to have this squashed together into one single commit.

But I tend to not repeat squash instructions as those always seem to fail for you.

pteeson commented:
I think understand what I need to do to rebase: 
git rebase -i HEAD~5 is what I am thinking of doing - replacing pick with squash - maybe making a minor edit to the most recent commit comment.
But……..Chapter 7 of the manual states

"Don’t include any commit you’ve already pushed to a central server – doing so will confuse other developers by providing an alternate version of the same change.” 

Since these 5 commits are already pushed to my pteeson/xbmc fork I am pausing. Am I incorrect in thinking of it as a “central server”? 
Of course xvmc/xbmc is a “central server”.

> On Oct 19, 2018, at 2:40 PM, Memphiz <notifications@github.com> wrote:
> Well the resulting readme is fine but that history with 5 commits is not. We still would like to have this squashed together into one single commit.
> But I tend to not repeat squash instructions as those always seem to fail for you.
> 
You are totally correct not to repeat. I’d rather learn how to fish! 
You have been most gracious in your support. Chapter 7 of the manual is helpful as well as the git online help.

kib commented:
Just force push to your own github. That will confuse noone as noone is forking from you. The PR will update automatically and can then be merged.

pteeson committed:
per email from Memphis re step 6.1 generate Xcode project - following commits will be squashed

pteeson committed:
Merge branch 'macOS-README-6.1' of github.com:pteeson/xbmc into macOS-README-6.1

pteeson commented:
It was getting horribly messy to the point where I deleted the former branch and created a new one - macOS-README-6.1-fix
Pushed the fix and created a pull request. Hope this is a little bit better.

> On Oct 20, 2018, at 4:13 PM, Kib <notifications@github.com> wrote:
> Just force push to your own github. That will confuse noone as noone is forking from you. The PR will update automatically and can then be merged.
